app/src/main/java/org/mozilla/fenix/components/Components.kt

```diff
@@ -71,14 +73,29 @@ class Components(private val context: Context) {
     }
 
     val addonCollectionProvider by lazy {
-        if (!BuildConfig.AMO_COLLECTION.isNullOrEmpty()) {
+        // Check if we have a customized (overridden) AMO collection (only supported in Nightly)
+        if (Config.channel.isNightlyOrDebug &&
+            (context.settings().overrideAmoCollection.isNotEmpty() ||
+                context.settings().overrideAmoUser.isNotEmpty())
+        ) {
+            AddonCollectionProvider(
+                context,
+                core.client,
+                collectionUser = context.settings().overrideAmoUser,
+                collectionName = context.settings().overrideAmoCollection
+            )
+        }
+        // Use build config otherwise
+        else if (!BuildConfig.AMO_COLLECTION.isNullOrEmpty()) {
             AddonCollectionProvider(
```

Comment:
nit: this could be simplified by using an internal function like:

```kotlin
fun createAddonCollectionFromBuildConfig() : AddonCollectionProvider{
 return  AddonCollectionProvider(
                context,
                core.client,
                collectionName = BuildConfig.AMO_COLLECTION,
                maxCacheAgeInMinutes = DAY_IN_MINUTES
            )
}
```